src/node/mod.rs

```diff
@@ -12,6 +12,76 @@
 // If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 //
 
+//! Simplicity Program Nodes
+//!
+//! The types in this module are used to represent Simplicity expressions as
+//! DAGs. The nodes of this DAG are individual combinators along with some
+//! cached data about the node, which depends on the specific node type.
+//!
+//! All nodes represent the root of an expression. Expressions whose source
+//! and target types are both unit are called "programs". Generally speaking,
+//! any nodes that users hold directly will be programs.
+//!
+//! There are three main node types:
+//!
+//! 1. [`ReedemNode`] represents a Simplicity node as it exists on the blockchain.
+//!    Every witness node is populated with a correctly-typed [`Value`], every
+//!    disconnect node has a child, and every non-executed branch of a `case`
+//!    combinator is replaced by a hidden node, and there is nothing you can do
```

Comment:
bebad27e1959bbd1f6e7229676035e139ae82fb7: If hidden nodes are a transport-layer thing now, I would avoid mentioning them. Maybe "every case is converted into a left or right assertion if only the left or right branch, respectively, is executed"

Reply:
Yep -- I should actually probably just replace this line with a TODO since we haven't implemented anything that would require full pruning. (If we added a fourth `UnprunedNode`, which you had to run the executor to get a `RedeemNode` from, that would achieve that. Thoughts?)